app/models/csv_export.rb

```diff
@@ -0,0 +1,47 @@
+class CsvExport < ActiveRecord::Base
+  belongs_to :user
+  serialize :filters, JSON
+  STATUS_VALUES = ['pending', 'started', 'finished', 'downloaded', 'failed', 'deleted']
+
+  before_destroy :delete_file
+
+  validates :status, inclusion: { in: STATUS_VALUES }
+  delegate :email, to: :user, allow_nil: true
+
+
+  scope :old, lambda {
+    end_date = Rails.application.config.samson.export_job.downloaded_age.seconds.ago
+    timeout_date = Rails.application.config.samson.export_job.max_age.seconds.ago
+    where("(status = 'downloaded' AND updated_at <= :end_date) OR created_at <= :timeout_date",
+      end_date: end_date, timeout_date: timeout_date)
+  }
+
+  def status?(state)
+    'ready' == state.to_s ? ['downloaded', 'finished'].include?(status) : status == state.to_s
+  end
+
+  def download_name
+    "deploys_#{created_at.to_s(:number)}.csv"
+  end
+
+  def path_file
+    "#{Rails.root}/export/#{id}"
+  end
+
+  def filters
+    filter = super.clone
+    if filter['deploys.created_at']
+      dates = filter['deploys.created_at'].scan(/-?\d+-\d+-\d+/)
```

Comment:
ahh that makes sense ... at least this way the data coming in/out of csv_export is the same :)